flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -420,27 +420,41 @@ void ArrayCopyAnalysis::arrayMentions(
   loadMapSets.insert({load, visited});
 }
 
+static bool hasPointerAttribute(mlir::Type type) {
```

Comment:
This name is confusing. I know it is a POINTER attribute in the surface syntax of Fortran, but at this point the syntax is long gone and we're in MLIR. In MLIR an Attribute is something completely different. `hasPtrType` or `hasPointerType` are better names.